### PR TITLE
add rpc param conversion to the server side execute

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

File: qa/rpc-tests/wallet.py [hunk 1]
```diff
@@ -284,16 +284,12 @@ def run_test (self):
...
             raise AssertionError("Must not parse invalid amounts")
 
-
-        try:
-            self.nodes[0].generate("2")
-            raise AssertionError("Must not accept strings as numeric")
-        except JSONRPCException as e:
-            assert("not an integer" in e.error['message'])
+        # this is now a valid call because we convert on server side instead of client side
+        self.nodes[0].generate("2")
```
Comment (gandrewstone) [inline qa/rpc-tests/wallet.py:292]: try a non-numerical string...

File: src/bitcoin-cli.cpp [hunk 2]
```diff
@@ -62,7 +62,12 @@ int CommandLineRPC(int argc, char *argv[])
         if (args.size() < 1)
             throw runtime_error("too few parameters (need at least command)");
         std::string strMethod = args[0];
-        UniValue params = RPCConvertValues(strMethod, std::vector<std::string>(args.begin() + 1, args.end()));
+        UniValue params(UniValue::VARR);
```
Comment (gandrewstone) [inline src/bitcoin-cli.cpp:65]: Are you forcing everything from the CLI to come in as a string here?  I'd prefer the CLI send the correct type, I just want the server to convert from a string if it is not...

EDIT: NVM, I see that the API doesn't even exist on the client anymore, so the only choice is to send them as strings

File: src/qt/rpcconsole.cpp [hunk 3]
```diff
@@ -226,8 +226,13 @@ void RPCExecutor::request(const QString &command)
         std::string strPrint;
         // Convert argument list to JSON objects in method-dependent way,
         // and pass it along with the method name to the dispatcher.
-        UniValue result = tableRPC.execute(
-            args[0], RPCConvertValues(args[0], std::vector<std::string>(args.begin() + 1, args.end())));
+        UniValue params(UniValue::VARR);
```
Comment (gandrewstone) [inline src/qt/rpcconsole.cpp]: as above

File: src/bitcoin-cli.cpp (hunk 2 above)
Comment (Greg-Griffith) [reply src/bitcoin-cli.cpp:65]: everything from cli is always a string, before cli would convert it before sending. now it just goes ahead and sends the string and the server side converts it

**Comment by Greg-Griffith** [comment]
@gandrewstone here is a better explaination of what the PR is doing since all of your comments seem to be along the same line.

Behavior without PR:
when a user enters a command into cli everything is a string, before sending to the server, the cli checks the conversion list because it "knows" some rpc call params are not supposed to be strings, so it converts and then sends.

something like python is does not go through cli so it is expected to send the correct types without referencing the convert table since it cant see it. 

the server expects everything to be the correct type already.

Behavior with PR:
the server now validates param types using the lookup table and no longer expects everything to be the correct type already. additionally, cli no longer checks the convert table. it just sends all the args as strings to server. 

if before a param would have been converted by cli, the server will now convert it after receiving regardless of where it comes from. so for the generate example you commented on above, the rpc call expects a number. so if python sends a number to the server, it will work fine. if it sends a string, since that function is in the convert table, the server will convert it to a number.

**Pull request merged**